internal/resources/grafana/resource_alerting_rule_group.go

```diff
@@ -198,9 +204,30 @@ func readAlertRuleGroup(ctx context.Context, data *schema.ResourceData, meta int
 		return err
 	}
 
-	if err := packRuleGroup(resp.Payload, data); err != nil {
-		return diag.FromErr(err)
+	g := resp.Payload
+	data.Set("name", g.Title)
+	data.Set("folder_uid", g.FolderUID)
+	data.Set("interval_seconds", g.Interval)
+	allowEditingFromUI := true
+	rules := make([]interface{}, 0, len(g.Rules))
+	for _, r := range g.Rules {
+		ruleResp, err := client.Provisioning.GetAlertRule(r.UID) // We need to get the rule through a separate API call to get the provenance.
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		r := ruleResp.Payload
+		data.Set("org_id", strconv.FormatInt(*r.OrgID, 10))
+		packed, err := packAlertRule(r)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if r.Provenance != "" {
+			allowEditingFromUI = false
```

Comment:
If at least one rule doesn't have Provenance set, do we completely disallow ui edition?

Reply:
Yeah, it'll mean that Terraform sees the whole rule group as not editable, and there will be a diff. On apply, all rules in the group will be updated to be the same (as configured)